### DISCUSSION TITLE
Plugins to exchange cryptos when received

### DISCUSSION THREAD

**Discussion by trymeouteh** [btcpayserver/btcpayserver#3627] [Ideas / Feature Requests]
To allow plugins to be integrated into BTCPay which will convert a crypto when received to another crypto the store wishes to hold instead.

For example, Bob buys a coffee with LTC, the store does not want to hold LTC but wants to accept LTC for payments. Once the coffeeshop receives the LTC, BTCPay will use a plugin if setup to exchange the LTC to another cryptocurrency such as BTC.

This can also increase privacy for the merchant and customer since each LTC transaction stores the LTC in a different address, buy converting the LTC right away which moves all the LTC from that single address will make it hard for anyone to trace the transaction since the other way of doing it is for the merchant to exchange all of his LTC from multiple addresses (merging them into one) in exchange for BTC. By merging addresses, it will show on the blockchain that everyone who sent LTC to those different LTC addresses, that all of those addresses are controlled by the same wallet.

**Comment by Kukks** [comment]
The fixedfloat/sideshift plugins do exactly this
